### PR TITLE
Refactor, unit and integration tests, doc

### PR DESCRIPTION
**Pull request by pandatix**
In this PR, there is a big refactor made for interconnectability of this wrapper with different implementation of an HTTP Client (through the interface `gonion.HTTPClient`).
This approach makes possible the unit and integration tests, but also using a custom implementation of an HTTP client with a more complex strategy (with request caching, or even a circuit breaking implementation).
Notice it removes the client structure while it does not implement a strategy (like handling a token or setting up the server URL in the requests). It decreases the shadowing of the wrapper.

While there is now unit tests with a good code coverage, you could add a badge in the `README.md` (I suggest using codecov.io) to promote the module.

To sum up what it embeds:
 - update Go to 1.17
 - interface for the HTTP Client improves interoperability
 - refactor the parameters structure handling using `gorilla/schema` module
 - refactor wrapper functions according to new client and parameters
 - unit tests (100% code coverage)
 - integration tests (discovered undocumented fields)
 - doc update and clean up

### CODE REVIEW

**Comment by R4yGM** [comment]
That's a pretty good and big pull request, everything seems to work fine and well to me.
Thank you for your contribute!
